site/en/docs/extensions/mv3/match_patterns/index.md

```diff
@@ -1,50 +1,161 @@
+- **host**: A hostname (`www.example.com`), a `*` before the hostname to match subdomains (`*.example.com`), or just a wildcard `*`.
 
-Here are some examples of _invalid_ pattern matches:
+- **path**: Must start with `/` and be present. If alone, it is treated as `/*`. For example: `/*`, `/foo*`, or `/foo/bar`. Each '`*`' matches 0 or more characters.
 
-<table class="fixed-table width-full"><tbody><tr><th style="margin-left:0; padding-left:0">Bad pattern</th><th style="margin-left:0; padding-left:0">Why it's bad</th></tr><tr><td><code>https://www.google.com</code></td><td>No <em>path</em></td></tr><tr><td><code>https://*foo/bar</code></td><td>'*' in the <em>host</em> can be followed only by a '.' or '/'</td></tr><tr><td><code>https://foo.*.bar/baz&nbsp;</code></td><td>If '*' is in the <em>host</em>, it must be the first character</td></tr><tr><td><code>http:/bar</code></td><td>Missing <em>scheme</em> separator ("/" should be "//")</td></tr><tr><td><code>foo://*</code></td><td>Invalid <em>scheme</em></td></tr></tbody></table>
+## Special cases {: #special }
 
-Some schemes are not supported in all contexts.
+`<all_urls>`
+: It matches any URL that starts with a permitted scheme, including any pattern listed under [valid patterns](#valid-examples). This is a broad host permission, meaning the Chrome web store [review may take longer](/docs/webstore/review-process/#review-time-factors).
 
-[1]: /docs/extensions/mv3/declare_permissions#host-permissions
-[2]: /docs/extensions/mv3/content_scripts
+`file:///`
+: Allows your extension to run on local files. It has no host and requires the user to manually [grant access][permissions-allow-access].
+
+Localhost URLs and IP addresses
+: To match any localhost port during development, use `http://localhost:*/*`. For example, it will match any paths under `http://localhost:8080` or `http://localhost:3000`. Another approach is to use `http://*:*/*` which will match any port. For IP addresses, specify the address plus a wildcard in the path. For example: `http://127.0.0.1/*`. 
```

Comment:
```suggestion
: To match any localhost port during development, use `http://localhost:*/*`. For example, it will match any paths under `http://localhost:8080` or `http://localhost:3000`.  For IP addresses, specify the address plus a wildcard in the path. For example: `http://127.0.0.1/*`. Another approach is to use `http://*:*/*` which will match localhost, IP addresses and any port.
```